src/input/keyboard.c

```diff
@@ -677,7 +686,8 @@ keyboard_key_notify(struct wl_listener *listener, void *data)
 				&& event->state == WL_KEYBOARD_KEY_STATE_PRESSED) {
 			start_keybind_repeat(seat->server, keyboard, event);
 		}
-	} else if (!input_method_keyboard_grab_forward_key(keyboard, event)) {
+	} else if (pressed_sent_key_is_released
```

Comment:
I find this logic hard to follow. Could we considering refactoring it with the aim of keeping `keyboard_key_notify()` easy on the brain. For example:

1. Could the `pressed_sent_key_is_released` logic go at the top of `input_method_keyboard_grab_forward_key()`.
2. Or maybe a `ime_grab()` wrapper containing both.